wgpu-hal/src/vulkan/device.rs

```diff
@@ -565,6 +583,12 @@ impl super::Device {
             .clipped(true)
             .old_swapchain(old_swapchain);
 
+        let mut format_list_info = vk::ImageFormatListCreateInfo::builder();
+        if !raw_view_formats.is_empty() {
+            format_list_info = format_list_info.view_formats(&raw_view_formats);
+            info = info.push_next(&mut format_list_info);
+        }
```

Comment:
Can we move the builder inside the if?

Reply:
Moving into `if` causes `format_list_info` to have a shorter lifetime than `info`

Reply:
Hmm, I see.